src/Microsoft.OData.Core/Metadata/MetadataUtils.cs

```diff
@@ -172,8 +172,10 @@ internal static IEdmOperation[] CalculateBindableOperationsForType(IEdmType bind
                 throw new ODataException(Strings.MetadataUtils_CalculateBindableOperationsForType(bindingType.FullTypeName()), exc);
             }
 
+            operations.EnsureOperationsBoundWithBindingParameter();

```

Comment:
Rather than looping through the operations here, and again in line 178, what if we changed EnsureOperationBoundWithBindingParameter() to take a single operation, and called it within the loop on line 178?